pkg/apis/serving/v1beta1/component.go

```diff
@@ -83,6 +84,18 @@ type ComponentExtensionSpec struct {
 	// Knative Pod Autoscaler(https://knative.dev/docs/serving/autoscaling/autoscaling-metrics).
 	// +optional
 	ScaleMetric *ScaleMetric `json:"scaleMetric,omitempty"`
+	// Type of metric to use. Options are Utilization, or AverageValue.
+	// +optional
+	ScaleMetricType *v2.MetricTargetType `json:"scaleMetricType,omitempty"`
+	// Address of Prometheus server.
+	// +optional
+	ServerAddress string `json:"serverAddress,omitempty"`
```

Comment:
Also consider adding QueryBackend which can be either prometheus or graphite, this can also be in the metrics configmap as these are mostly global config